PEAR-1927: Responsive summary pages

paribartandhakal opened the pull request:
## Description

Also fixed:
https://gdc-ctds.atlassian.net/browse/PEAR-2089
https://gdc-ctds.atlassian.net/browse/PEAR-1908

## Checklist
- [N/A] Added proper unit tests
- [N/A] Left proper TODO messages for any remaining tasks
- [N/A] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

## Screenshots/Screen Recordings (if Appropriate)

**_So many pages_**

Commit:
file summary page

Commit:
case summary changes

Commit:
ssms and gene summary page

Commit:
project summary page - responsive changes

Commit:
last few style changes - code cleanup

Commit:
move export to correct file

paribartandhakal commented on packages/portal-proto/src/components/Summary/SummaryHeader.tsx:
```diff
@@ -19,41 +25,87 @@ export const SummaryHeader = ({
+
+  useEffect(() => {
+    const globalHeader = document.querySelector("#global-header");
+    const resizeObserver = new ResizeObserver((entries) => {
```
Probably can't use mantine's `useResizeObserver` because we need to select an element from outside the component.

amlehman replied:
What is this block of code doing? I wasn't really able to figure it out from looking at the page/code.

amlehman commented on packages/portal-proto/src/features/projects/SummaryHeaderLeft.tsx:
```diff
@@ -0,0 +1,253 @@
+import { FiDownload as DownloadIcon } from "react-icons/fi";
+import { ProjectViewProps } from "./ProjectView";
+
+function SummaryHeaderLeft({ projectData }: { projectData: ProjectViewProps }) {
```
Maybe `SummaryHeaderControls` is a better name for this component? I spent a while trying to figure out if this was a component that displayed on the left at a breakpoint.

paribartandhakal replied:
Sure

amlehman commented on packages/portal-proto/src/utils/index.ts:
```diff
@@ -284,3 +284,5 @@ export const statusBooleansToDataStatus = (
 export const focusStyles =
   "focus-visible:outline-none focus-visible:ring-offset-2 focus:ring-offset-white rounded-md focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-focusColor";
+
+export const LG_BREAKPOINT = 1024;
```
Can we import this from the tailwind theme? I think its a good idea to just have one source of truth for what the breakpoint values are.

`export const LG_BREAKPOINT = parseInt(tailwindConfig.theme.extend.screens.lg)`

paribartandhakal replied:
I agree. Thank you for mentioning.

Commit:
review comments

paribartandhakal commented:
@amlehman I couldn't reply to your comment above re. the `useEffect ` code.

`SummaryHeader` is fixed just below the global header in the entity summary pages. Previously, I had used `fixed` class which made its position absolute. As a result the actual summary pages were given some random margin top to be shown below the `SummaryHeader`.

Now I changed the `SummaryHeader` to use `sticky` so that we don't have to use random margin tops. For this reason I am calculating the height of the global header during resize so that I can give `SummaryHeader` a correct top offset and use a fixed margin tops for all the summary pages.

Commit:
add removed styl

Commit:
fix some styling - responding to cemile's comments

wteouchicago commented:
1. The Search/Expand All section of the Biospecimen tree now differs from develop and the mockups:
<img width="678" alt="Screenshot 2024-08-30 at 10 34 23 AM" src="https://github.com/user-attachments/assets/01c81a5a-2633-4e57-a5fb-ec95642606e0">

2. May be a preexisting issue. File summary page > Associated cases/biospecimens table > the Total of and search area seems weirdly aligned with the title of the section (at least compared with the alignment of the Read Groups section below). Try this file: http://localhost:3000/files/4600677b-4371-44cd-8e55-818f271e84f2
<img width="1110" alt="Screenshot 2024-08-30 at 10 55 01 AM" src="https://github.com/user-attachments/assets/9d1db3ee-a0e0-4ff3-bfab-c97b559b949a">

3. Is this intended -- the non-alignment of the summary tables due to the titles requiring different number of rows? (It's fine if we don't address since it doesn't happen all the time. Just didn't know if we tried to address and missed some situations.)

<img width="1186" alt="Screenshot 2024-08-30 at 11 01 42 AM" src="https://github.com/user-attachments/assets/7b14f2fa-565d-4cb9-bc77-f5007284f120">

Commit:
fix biospeciment and table header alignment

paribartandhakal commented:
> 1. The Search/Expand All section of the Biospecimen tree now differs from develop and the mockups:
> 
> <img alt="Screenshot 2024-08-30 at 10 34 23 AM" width="678" src="https://private-user-images.githubusercontent.com/73256434/363236334-01c81a5a-2633-4e57-a5fb-ec95642606e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzcyMTQsIm5iZiI6MTcyNTAzNjkxNCwicGF0aCI6Ii83MzI1NjQzNC8zNjMyMzYzMzQtMDFjODFhNWEtMjYzMy00ZTU3LWE1ZmItZWM5NTY0MjYwNmUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDE2NTUxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0ODg4ZmI4OWNkZDE2MTAzNDE4NWI5MzVjOWU4M2I4N2Q1ZjRhN2I4OTM2NTAwZWI4Zjk5NjRmYWUwMzVhZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.libNV30NEaP84J6hcizNN0Xf-F644vhUCnytKtD4TW4">
> 2. May be a preexisting issue. File summary page > Associated cases/biospecimens table > the Total of and search area seems weirdly aligned with the title of the section (at least compared with the alignment of the Read Groups section below). Try this file: http://localhost:3000/files/4600677b-4371-44cd-8e55-818f271e84f2
> 
> <img alt="Screenshot 2024-08-30 at 10 55 01 AM" width="1110" src="https://private-user-images.githubusercontent.com/73256434/363243682-9d1db3ee-a0e0-4ff3-bfab-c97b559b949a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzcyMTQsIm5iZiI6MTcyNTAzNjkxNCwicGF0aCI6Ii83MzI1NjQzNC8zNjMyNDM2ODItOWQxZGIzZWUtYTBlMC00ZmYzLWJmYWItYzk3YjU1OWI5NDlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDE2NTUxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiM2I1MDlhM2EwZmY4NTQ2ZjFkMDM2NjI2OTdjOWEyOWM5NjI1NWFkMzRhOWIzNDIyZDljZWExN2FjOTkzYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dGkCq6f-FcuLZAxjmAXEPmY8j0PZ0xO8NDeE7-iFHJY">
> 3. Is this intended -- the non-alignment of the summary tables due to the titles requiring different number of rows? (It's fine if we don't address since it doesn't happen all the time. Just didn't know if we tried to address and missed some situations.)
> 
> <img alt="Screenshot 2024-08-30 at 11 01 42 AM" width="1186" src="https://private-user-images.githubusercontent.com/73256434/363265021-7b14f2fa-565d-4cb9-bc77-f5007284f120.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzcyMTQsIm5iZiI6MTcyNTAzNjkxNCwicGF0aCI6Ii83MzI1NjQzNC8zNjMyNjUwMjEtN2IxNGYyZmEtNTY1ZC00Y2I5LWJjNzctZjUwMDcyODRmMTIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDE2NTUxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmZmIwOThjZjMyMjIwZTYyZWFjMGI0ZTg2NWU0Mjg4NmU3MjdkNGU5MmNiYzI4NzZiNDE1ZDgxMDBkMDI2YzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kEWEdAOC-W68zz_bdHPk_twTAjH1PQbmw1lyezBTcZ0">

1. I overlooked # 1. I just fixed it.
2. The alignment issue is now fixed.
3. I had noticed it but I left it as it is for now. Not sure what to do in there.

wteouchicago commented:
1 & 2 look good. 👍 

Let's pretend 3 never happened. ;) 👍
